Language/Haskell/GhcMod/Monad.hs

```diff
@@ -52,17 +52,17 @@ import Exception
 import System.Directory
 import Prelude
 
-withGhcModEnv :: (IOish m, GmOut m) => FilePath -> Options -> (GhcModEnv -> m a) -> m a
+withGhcModEnv :: (IOish m, GmOut m) => FilePath -> Options -> ((GhcModEnv, GhcModLog)  -> m a) -> m a
 withGhcModEnv = withGhcModEnv' withCradle
  where
    withCradle dir =
-       gbracket (findCradle' dir) (liftIO . cleanupCradle)
-
-withGhcModEnv' :: (IOish m, GmOut m) => (FilePath -> (Cradle -> m a) -> m a) -> FilePath -> Options -> (GhcModEnv -> m a) -> m a
+       gbracket (runJournalT $ findCradle' dir) (liftIO . cleanupCradle . fst)
```

Comment:
Um the log level is
```
Just b `mplus` Just a = Just b
```
This way a `Nothing` is always overidden by a `Just` and once it's `Just` it can't become `Nothing`:

```
Nothing `mplus` Just a = Just a
```

(mplus has the arguments the other way around compared to mappend)